src/cat.jl

```diff
@@ -24,23 +23,25 @@ struct ConcatDiskArray{T,N,P,C,HC,ID} <: AbstractDiskArray{T,N}
     innerdims::Val{ID}
 end
 
-function ConcatDiskArray(arrays::AbstractArray{Union{<:AbstractArray,Missing}})
+function ConcatDiskArray(arrays::AbstractArray{Union{<:AbstractArray,Missing}}; fill=missing)
     et = Base.nonmissingtype(eltype(arrays))
-    T = Union{Missing,eltype(et)}
+    T = promotetype(typeof(fill), eltype(et))
```

Comment:
I thought the passed in array would still contain either arrays or `missing` values, but we would just return `fill` for those sections, so `fill` would need to be a struct field.

So what we have here is kinda half way. Probably we should either drop the `fill` keyword completely or keep it as a struct field and use `missing` in the array.

Reply:
I prefer to drop the fill keyword. I actual thought I already did so. 
With my approach we can have different fill values in different parts of the concatenated array.

Reply:
But is that actually useful? I can't imagine a real use case for multiple fill values. And putting `missing` in the array seems clearer for representing missing arrays. 

The missing value used in the array of arrays doesn't have to be coupled to the fill value.

Reply:
I agree that it might be a bit dangerous to allow arbitrary values to represent missings. OTOH one never knows when this might become useful for some reason. I am ok with both ways and would currently slightly prefer @felixcremer s solution because there is already an implementation. If we want to be a bit more on the safe side, we could introduce a small wrapper type like

````julia
struct MissingTile{F}
    fillvalue::F
end
````
that explicitly marks missing entries to be concatenated and how they should be filled. The main reason why the current solution might be problematic is for DiskArrays with arrays as eltype, but maybe this is overthinking things a bit too much.